src/main/java/com/parsing/parsers/pdf/parsing/RectangleDetector.java

```diff
@@ -0,0 +1,231 @@
+        for (double[] line : lines) {
+            pointsY.add(line[1]);
+        }
+        return pointsY.stream().sorted().toList();
+    }
+
+    private double[] sortListArray(List<double[]> lines) {
+        double minY = 50;
+        double maxY = 0;
+        double averageX = 0;
+        for (double[] doubles : lines) {
+            if (minY > doubles[1]) {
+                minY = doubles[1];
+            } else if (maxY < doubles[1]) {
+                maxY = doubles[1];
+            }
+            if (minY > doubles[3]) {
+                minY = doubles[3];
+            } else if (maxY < doubles[3]) {
+                maxY = doubles[3];
+            }
+            averageX = (averageX + doubles[0]) / 2;
+        }
+        return new double[]{averageX, maxY, averageX, minY};
+    }
+
+    private List<Double> sortLinesByX(List<double[]> lines) {
+        List<Double> result = new ArrayList<>();
+        for (double[] line : lines) {
+            result.add(line[0]);
+        }
+        return result.stream().sorted().toList();
+    }
+
+    private HorizontalLineCoordinate findExtremeHorizontalTablePoints(List<double[]> lines) {
+        double x1 = 100;
+        double x2 = 0;
+        for (double[] array : lines) {
+            if (array[0] < x1) {
+                x1 = array[0];
+            }
+            if (array[2] > x2) {
+                x2 = array[2];
+            }
+        }
+        return new HorizontalLineCoordinate(x1, x2, 0);
+    }
+
+    public List<double[]> findVerticalLinesWithOpenCV(String fileSource) {
+        Mat dst = new Mat();
+        Mat cdst = new Mat();
+        OpenCV.loadLocally();
+        Mat source = Imgcodecs.imread(fileSource, Imgcodecs.IMREAD_GRAYSCALE);
+        if (source.empty()) {
+            log.warn("Error opening image! Program Arguments: [image_name -- default " + fileSource + "] \n");
+            System.exit(-1);
+        }
+        Imgproc.Canny(source, dst, 50, 200, 3, false);
+        Imgcodecs.imwrite("afterCannyRect.png", dst);
+        Imgproc.cvtColor(dst, cdst, Imgproc.COLOR_GRAY2BGR);
+        verticalLinesMat = cdst.clone();
+        Mat linesP = new Mat();
+        Imgproc.HoughLinesP(dst, linesP, 3, Math.PI, 200, RectangleDetector.VERTICAL_LINE_LENGTH, 5);
+        List<double[]> lines = new ArrayList<>();
+        for (int x = 0; x < linesP.rows(); x++) {
+            double[] l = linesP.get(x, 0);
+            lines.add(l);
+        }
+        Imgcodecs.imwrite("verticalLines.png", linesP);
+        return lines;
+    }
+
+    private List<double[]> findHorizontalLinesWithOpenCV(String fileSource) {
+        Mat dst = new Mat();
+        Mat cdst = new Mat();
+        OpenCV.loadLocally();
+        Mat source = Imgcodecs.imread(fileSource, Imgcodecs.IMREAD_GRAYSCALE);
+        if (source.empty()) {
+            log.warn("Error opening image! Program Arguments: [image_name -- default " + fileSource + "] \n");
+            System.exit(-1);
```

Comment:
SYstem.exit should be removed. We don't need to stop an app because of it.

Reply:
Yes, I deleted it.